### PR TITLE
[timeseries] Avoid importing pytorch_lightning at top level

### PR DESCRIPTION
**Pull request by shchur**
A recent update to `pytorch_lightning` has potential to introduce a massive performance regression to forecasting models in AutoGluon that rely on multiprocessing / joblib. As of `v2.0.8` of `pytorch_lightning`, a multiprocessing lock is created whenever the module is imported. This increased the process initialization time significantly.

For example, total runtime of the above code snippet
```python
import pytorch_lightning as pl
from autogluon.timeseries import TimeSeriesPredictor, TimeSeriesDataFrame

data = TimeSeriesDataFrame('https://autogluon.s3.amazonaws.com/datasets/timeseries/m4_hourly/train.csv')
predictor = TimeSeriesPredictor()
predictor.fit(data, hyperparameters={"Naive": {}})
```

- current `master` branch: 1m 42s
- with this PR: 36s.

*Description of changes:*
- Hide all imports of `pytorch_lightning` and `torch` inside class methods. This ensures that they are not imported by the processes spawned by joblib.
- Create a separate method for setting the random seed that does not rely on `pytorch_lightning`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: common/src/autogluon/common/utils/utils.py [hunk 1]
```diff
@@ -176,3 +177,14 @@ def hash_pandas_df(df: Optional[pd.DataFrame]) -> str:
...
+
+def seed_everything(seed: int) -> None:
+    """Set random seeds for numpy and PyTorch."""
+    np.random.seed(seed)
+    try:
+        import torch
+
+        torch.manual_seed(seed)
+    except ImportError:
+        pass
```
Comment (shchur) [inline common/src/autogluon/common/utils/utils.py:190]: @yinweisu do we have this functionality anywhere else in `core` or `common`, or is it fine to add it here?

Comment (yinweisu) [reply common/src/autogluon/common/utils/utils.py:190]: I think it's fine to add it here

**Review by melopeo — APPROVED** [review]
